internal/rbd/rbd_attach_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestParseMapOptions(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name              string
+		mapOption         string
+		expectKrbdOptions string
+		expectNbdOptions  string
+		expectErr         string
+	}{
+		{
+			name:              "with old format",
+			mapOption:         "kOp1,kOp2",
+			expectKrbdOptions: "kOp1,kOp2",
+			expectNbdOptions:  "",
+			expectErr:         "",
+		},
+		{
+			name:              "with new format",
+			mapOption:         "krbd:kOp1,kOp2;nbd:nOp1,nOp2",
+			expectKrbdOptions: "kOp1,kOp2",
+			expectNbdOptions:  "nOp1,nOp2",
+			expectErr:         "",
+		},
+		{
+			name:              "without krbd: label",
+			mapOption:         "kOp1,kOp2;nbd:nOp1,nOp2",
+			expectKrbdOptions: "kOp1,kOp2",
+			expectNbdOptions:  "nOp1,nOp2",
+			expectErr:         "",
+		},
+		{
+			name:              "with only nbd label",
+			mapOption:         "nbd:nOp1,nOp2",
+			expectKrbdOptions: "",
+			expectNbdOptions:  "nOp1,nOp2",
+			expectErr:         "",
+		},
+		{
+			name:              "unknown mounter used",
+			mapOption:         "xyz:xOp1,xOp2",
+			expectKrbdOptions: "",
+			expectNbdOptions:  "",
+			expectErr:         "unknown mounter type",
+		},
+		{
+			name:              "bad formatted options",
+			mapOption:         "nbd:nOp1:nOp2;",
+			expectKrbdOptions: "",
+			expectNbdOptions:  "",
+			expectErr:         "badly formatted map/unmap options",
+		},
+	}
+	for _, tt := range tests {
+		tc := tt
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			krbdOpts, nbdOpts, err := parseMapOptions(tc.mapOption)
+			if err != nil && !strings.Contains(err.Error(), tc.expectErr) ||
+				krbdOpts != tc.expectKrbdOptions || nbdOpts != tc.expectNbdOptions {
+				t.Errorf("parseMapOptions(%s) "+
+					"krbdOpts => [expected :%q, got: %q]; "+
+					"nbdOpts => [expected :%q, got: %q]; "+
+					"err => [exp: %v, got: %v];",
+					tc.mapOption,
+					tc.expectKrbdOptions, krbdOpts,
+					tc.expectNbdOptions, nbdOpts,
+					tc.expectErr, err)
```

Comment:
```suggestion
if err != nil && !strings.Contains(err.Error(), tc.expectErr) {
// log error
if krbdOpts != tc.expectKrbdOptions{
// log unmatched krbd option error
}
if nbdOpts != tc.expectNbdOptions{
// log unmatched nbd option error
}
```

This helps in debugging.

Reply:
I don't know how this makes a difference, If that is for better readability, so be it! Done now!

Reply:
it's more for debugging as unit test failure and for readability.